subsys/event_manager/event_manager.c

```diff
@@ -175,7 +160,9 @@ static void event_processor_fn(struct k_work *work)
 
 		const struct event_type *et = eh->type_id;
 
-		event_manager_trace_event_execution(eh, true);
+		STRUCT_SECTION_FOREACH(event_preprocess_hook, h) {
```

Comment:
maybe we cold have a enable/disable switch to opt out the loop?

Reply:
Would not it be optimized if section start equals section end?

Reply:
I don't think so. Only linker knows that and we don't have LTO. So the traverse code will execute (will load two pointers and do compare).

Reply:
maybe we can hide this behind a helper?
e.g.
```
if ENABLED
CALL_HOOKS(event_preprocess_hook) 
#else
CALL_HOOKS(event_preprocess_hook) STRUCT_SECTION_FOREACH(event_preprocess_hook, h) { h->hook(eh); };
```
or something nicer

Reply:
Oh man... I hate not having LTO enabled. And being forced to manually solve thinks that should be solved by the program. With LTO it could work without any configuration needed. OK then - I will add such an option.

Reply:
Done